hack/jenkins/upload-to-gcs.sh

```diff
@@ -151,15 +151,22 @@ function update_job_result_cache() {
 
   mkdir -p ${tmp_results%/*}
 
+  # Construct a valid json file
```

Comment:
so as currently written, it's not a valid JSON file (which allows only a single object at root), but `jq` seems to work:

input:
```
{"version": "v1.4.0-alpha.2.793+a4acf0874763b9", "buildnumber": "7478", "result": "SUCCESS"}
{"version": "v1.4.0-alpha.2.795+16c71aad4a2266", "buildnumber": "7479", "result": "SUCCESS"}
```

```shell
$ jq -r '. | select(.result == "SUCCESS") | select(.version != "") | [.version,.buildnumber] | "\(.[0]) \(.[1])"' /
tmp/out2.json
v1.4.0-alpha.2.793+a4acf0874763b9 7478
v1.4.0-alpha.2.795+16c71aad4a2266 7479
```

If you wanted it to be actually valid JSON for anything that reads it, then it could really just be an array:
```
[
        {"version": "v1.4.0-alpha.2.793+a4acf0874763b9", "buildnumber": "7478", "result": "SUCCESS"},
        {"version": "v1.4.0-alpha.2.795+16c71aad4a2266", "buildnumber": "7479", "result": "SUCCESS"},
        {}
]
```

```shell
$ jq -r '.[] | select(.result == "SUCCESS") | select(.version != "") | [.version,.buildnumber] | "\(.[0]) \(.[1])"'
 /tmp/out.json
v1.4.0-alpha.2.793+a4acf0874763b9 7478
v1.4.0-alpha.2.795+16c71aad4a2266 7479
```

I'm fine with the object wrapper you've implemented here too. Not sure if you use the "JOB" name anywhere else.